### ISSUE TITLE
Entering custom weights in multiplier menu is tedous

### ISSUE THREAD

**Issue by gyhtomit** [rozPierog/Cofi#229]
**Describe the bug**
Always requiring an integer part in the weight multiplier makes trying to enter custom weights rather tedious. E.g. adapting the default French Press recipe from 30.0g coffee to 45.0g coffee requires moving the cursor several times.
This happens because the app inserts a 0 as an integer part, if all digits left of the dot are deleted.

**To Reproduce**
1. Go to any recipe and open the multiplier menu (scale button, top right)
2. Place the cursor on the very right in the coffee multiplier field ("30.0|")
3. Repeatedly deleting digits by pressing the backspace button results in:
	1. "30.|0"
	2. "30|0.0"
	3. "3|0.0"
	4. "|0.0"

Alternatively placing the cursor to the left of the dot ("30|.0") and repeatedly pressing backspace results in a zero being inserted in the integer part:
1. "3|.0"
2. "|0.0"

**Expected behavior**
Adapting the weights should be as easy as
1. Deleting the current value
2. writing the desired value

**Screenshots**
none. Will provide if needed.

**Smartphone:**
 - Device: Pixel 4a
 - Android: 13
 - App Version: 1.20.0 (via F-Droid)

**Additional context**
To actually change the 30.0g to 45.0g I have to enter 45 before deleting the 30 which requires moving the cursor multiple times.

**Comment by KhaaL** [comment]
Appriciate this bug report - I jist downloaded the app and was bugged by the same thing 😊

**Comment by rozPierog** [comment]
Agreed that it's not very intuitive, and/or broken. I have little to no time for Cofi work right at this very moment, but I'll be happy to merge a PR if somebody wants to fix it!
